### ISSUE TITLE
Restarting after a crash no longer works

### ISSUE THREAD

**Issue by Mohamed00** [nvaccess/nvda#12557]
### Steps to reproduce:
1. Find a way to crash NVDA. I reproduced this reliably by crashing a synthesizer, but I don't think this is specific to that particular case.
### Actual behavior:
NVDA does not restart after the crash.
### Expected behavior:
NVDA should restart.
#### NVDA installed/portable/running from source:
Installed.
#### NVDA version:
alpha-23112,0d98162d
#### Windows version:
Windows 10, build 21390.2025
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/a
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, 2020.4 works normally. This popped up somewhere around the time NVDA's exit process was refactored, I believe.
#### If add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.

**Comment by XLTechie** [comment]
>  1. Find a way to crash NVDA. I reproduced this reliably by crashing a synthesizer, but I don't think this is specific to that particular case.
>   Actual behavior:
> NVDA does not restart after the crash.

What leads you to believe that NVDA should be expected to restart on its 
own after a runtime crash such as this?

When (under what version) did that ever work?

**Comment by Mohamed00** [comment]
This worked normally in 2020.4, but doesn't seem to work now. NVDA has 
been able to restart itself after some crashes since 2011.3.

**Comment by feerrenrut** [comment]
@Mohamed00 we really need specific details. Ideally also a log and crash dump.

**Comment by Mohamed00** [comment]
The crash occurred in the SAPI4 version of ViaVoice. It's a bug in the synthesizer, but NVDA was previously able to reliably restart after a crash like this. However, in recent builds, it doesn't restart anymore. 
[log and crash dump](https://github.com/nvaccess/nvda/files/6674332/)

**Comment by Mohamed00** [comment]
Sorry, accidentally broke the link. It's fixed now.

**Comment by seanbudd** [comment]
Fixed by #12562 (merged to alpha in #12572)
